tests/tests/classes/PluginManager.js

```
@@ -0,0 +1,299 @@
+      expect(events.length).to.be.at.least(3);
+    });
+
+    it('should get all the matching events for a nestec level command', () => {
+      const commandsArray = 'deploy onpremises'.split(' ');
+      const events = pluginManager._getEvents(commandsArray, pluginManager._commands);
+
+      // Note: We expect at least 3 because 3 events will be created for each lifeCycleEvent
+      expect(events.length).to.be.at.least(3);
+    });
+
+    it('should return an empty events array when the command is not defined', () => {
+      const commandsArray = 'nonExistingCommand'.split(' ');
+      const events = pluginManager._getEvents(commandsArray, pluginManager._commands);
+
+      expect(events.length).to.equal(0);
+    });
+  });
+
+  describe('#runCommand()', () => {
+    it('should not care about correct capitalization of function name inside a hook', () => {
+      class WrongCapitalizedHookPluginMock {
+        constructor() {
+          this.commands = {
+            deploy: {
+              usage: 'Deploy to the default infrastructure',
+              lifeCycleEvents: [
+                'functions',
+              ],
+            },
+          };
+
+          this.hooks = {
+            // should be "beforeFunctions" but the PluginManager should not bother
+            'deploy:beforefunctions': this.functions.bind(this)
+          };
+
+          // used to test if the function was executed correctly
+          this._deployedFunctions = 0;
+        }
+
+        functions() {
+          this._deployedFunctions = this._deployedFunctions + 1;
+        }
+      }
+
+      pluginManager._addPlugin(WrongCapitalizedHookPluginMock);
+
+      const command = 'deploy';
+      pluginManager.runCommand(command);
+
+      expect(pluginManager._pluginInstances[0]._deployedFunctions).to.equal(1);
+    });
+
+    it('should do nothing when the given command is not available', () => {
```

We might want to throw an error here, but not sure yet. @eahefnawy @pmuens how is this handled now? Another option would be to do nothing here and simply leave error handling to the CLI class, so a command that doesn't exist but is called here simply does nothing.